site/src/self_profile.rs

```diff
@@ -37,9 +40,15 @@ pub fn generate(
         ProcessorType::Crox => {
             let opt = serde_json::from_str(&serde_json::to_string(&params).unwrap())
                 .context("crox opts")?;
+            let data = crox::generate(self_profile_data, opt).context("crox")?;
```

Comment:
Probably most clients should support, but for correctness, we should only use GZIP if the HTTP request specifies that it accepts GZIP encoding in the `Accept-Encoding` header (there can be multiple options in it, hopefully some of ours HTTP crates can parse it).